### ISSUE TITLE
Drone Flying Backward & CUDA Error During Simulation

### ISSUE THREAD

**Issue by Liustt** [HenryHuYu/DiffPhysDrone#23]
When launching the simulation, the drone unexpectedly started flying backward, 
https://github.com/user-attachments/assets/004de38b-24ee-4f53-b57b-655d73428b2d, and the console printed the following error message: no kernel image is available for execution on the device 209

Solution：
1. Check your GPU’s compute capability (SM value). For example, the RTX 4060 has an SM value of 89.
2. Edit the CMake configuration file: src/agile_autonomy/data_generation/agile_autonomy/CMakeLists.txt, and locate the line containing CUDA_NVCC_FLAGS.
3. Add the following flag: -gencode=arch=compute_89,code=sm_89  (89 should be replaced by your GPU's SM code)
4. Rebuild the project.

**Comment by illfen** [comment]
Have you tried running it with your own trained model? I can fly normally with base.pth, but when I switch to my own trained model, it flies backward.

**Comment by Liustt** [comment]
The model used in Flightare is not exactly the same as the one in the training code. You can verify this, and a simple modification should make it compatible.

**Comment by illfen** [comment]
Could you please explain in more detail? Do I need to modify model.py so its layer names and structure match base.pth and then retrain?

**Comment by chnqiyuu** [comment]
PlannerBase.py中第226行难道不是本来就是act, self.h = self.model(depth, state, self.h)嘛，我改了model文件还是效果很差直接撞树，想问问你有修改他给的训练代码的参数吗

**Comment by Liustt** [comment]
之前的回答有误，如果使用默认参数自训练模型做仿真的话，可以这样改：首先model.py`import torch
from torch import nn

def g_decay(x, alpha):
    return x * alpha + x.detach() * (1 - alpha)

class Model(nn.Module):
    def __init__(self, dim_obs=9, dim_action=4) -> None:
        super().__init__()
        self.stem = nn.Sequential(
            nn.Conv2d(1, 32, 2, 2, bias=False),  # 1, 12, 16 -> 32, 6, 8
            nn.LeakyReLU(0.05),
            nn.Conv2d(32, 64, 3, bias=False), #  32, 6, 8 -> 64, 4, 6
            nn.LeakyReLU(0.05),
            nn.Conv2d(64, 128, 3, bias=False), #  64, 4, 6 -> 128, 2, 4
            nn.LeakyReLU(0.05),
            nn.Flatten(),
            nn.Linear(128*2*4, 192, bias=False),
        )
        self.dim_obs = dim_obs
        self.v_proj = nn.Linear(dim_obs, 192)
        self.v_proj.weight.data.mul_(0.5)

        self.gru = nn.GRUCell(192, 192)
        self.fc = nn.Linear(192, dim_action, bias=False)
        self.fc.weight.data.mul_(0.01)
        self.act = nn.LeakyReLU(0.05)

    def reset(self):
        pass

    def forward(self, x: torch.Tensor, v, hx=None):
        img_feat = self.stem(x)
        x = self.act(img_feat + self.v_proj(v))
        hx = self.gru(x, hx)
        act = self.fc(self.act(hx))
        return act, None, hx


if __name__ == '__main__':
    Model()`  之后PlannerBase.py第226行改为：act, values, self.h = self.model(depth, state, self.h)，我本地是可以运行的
